lib/importer/strategies/custom-importer.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+
+const fs = require('../../file-system'),
+      Container = require('aurelia-dependency-injection').Container,
+      logger = require('aurelia-logging').getLogger('Custom-importer');
+
+let CustomImporterStrategy = class {
+
+  static inject() { return [Container, 'package']; };
+
+  constructor(container, pkg) {
+    this.container = container;
+    this.package = pkg;
+  }
+
+  applies() {
+    let location = this.getStrategyLocation();
+
+    return fs.exists(location)
+    .then(available => {
+      if (available) {
+        let strategy = this.getStrategy();
+        let result = strategy.applies();
+
+        return result;
+      }
+
+      logger.debug(`The plugin does not have a custom importer module. Looked for "${location}"`);
+
+      return false;
+    });
+  }
+
+  execute() {
+    let strategy = this.getStrategy();
+    let result = strategy.execute();
+
+    return result;
+  }
+
+  getStrategy() {
+    let ctor = require(this.getStrategyLocation());
+    return this.container.get(ctor);
+  }
+
+  getStrategyLocation() {
+    return fs.resolve(fs.join(this.package.rootPath, 'importer-callbacks.js'));
```

Comment:
I must have missed this e-mail notification. I don't quite like the `-callbacks` part. Perhaps something like `import-handler.js`. Also, I wonder if we should reference this file from package.json so that plugin authors can decide how to name the file (and where it is located)